From: SeongJae Park <sj@kernel.org>
Subject: mm/memory_hotplug: return nothing from do_migrate_range()
Date: Wed, 15 Feb 2023 23:02:59 +0000

Patch series "mm/memory_hotplug: cleanup do_migrate_range()".

Make do_migrate_range() return value mechanism simple.


This patch (of 2):

Return value mechanism of do_migrate_range() is not very simple, while
no caller of the function checks the return value.  Make the function
return nothing to be more simple.

Link: https://lkml.kernel.org/r/20230215230300.61125-1-sj@kernel.org
Link: https://lkml.kernel.org/r/20230215230300.61125-2-sj@kernel.org
Suggested-by: David Hildenbrand <david@redhat.com>
Signed-off-by: SeongJae Park <sj@kernel.org>
Cc: Oscar Salvador <osalvador@suse.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memory_hotplug.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/mm/memory_hotplug.c~mm-memory_hotplug-return-nothing-from-do_migrate_range
+++ a/mm/memory_hotplug.c
@@ -1620,8 +1620,7 @@ found:
 	return 0;
 }
 
-static int
-do_migrate_range(unsigned long start_pfn, unsigned long end_pfn)
+static void do_migrate_range(unsigned long start_pfn, unsigned long end_pfn)
 {
 	unsigned long pfn;
 	struct page *page, *head;
@@ -1721,8 +1720,6 @@ do_migrate_range(unsigned long start_pfn
 			putback_movable_pages(&source);
 		}
 	}
-
-	return ret;
 }
 
 static int __init cmdline_parse_movable_node(char *p)
_
